packages/realm/src/Realm.ts

```diff
@@ -439,21 +451,21 @@ export class Realm {
    */
   constructor(config: Configuration);
   /** @internal */
-  constructor(config: Configuration, internal: binding.Realm);
-  /** @internal */
-  constructor(internal: binding.Realm, schemaExtras?: RealmSchemaExtra);
-  constructor(arg: Configuration | binding.Realm | string = {}, secondArg?: object) {
-    if (arg instanceof binding.Realm) {
-      this.schemaExtras = (secondArg ?? {}) as RealmSchemaExtra;
-      this.internal = arg;
-    } else {
-      const config = typeof arg === "string" ? { path: arg } : arg;
+  constructor(config: Configuration | null, internalConfig: InternalConfig);
+  constructor(arg?: Configuration | string | null, internalConfig: InternalConfig = {}) {
+    const config = typeof arg === "string" ? { path: arg } : arg || {};
+    // Calling `Realm.exists()` here is necessary to capture the correct value when
+    // the constructor was called since the realm may be opened during this construction.
+    // This is needed when deciding whether to update initial subscriptions.
```

Comment:
How about:
```ts
// Calling `Realm.exists()` before `binding.Realm.getSharedRealm()` is necessary to capture
// the correct value when the constructor was called since `binding.Realm.getSharedRealm()`
// will open the realm. This is needed when deciding whether to update initial subscriptions.
```